src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -19,92 +19,135 @@
-
-    public void ProcessRequest(HttpContext context)
-    {
-        // Check for prerequisites needed before this script can execute
+    private string _gitExecutablePath;
+    private string _gitRepositoriesBasePath;
+    private TimeSpan _gitExecutionTimeout;
 
-        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+    private bool _configurationRead = false;
 
-        StringBuilder result = new StringBuilder();
-        bool success = CheckPrerequisites(result);
+    private const string _gitExecutablePathConfigurationKeyString = "GitExecutablePath";
+    private const string _gitRepositoriesBasePathConfigurationKeyString = "GitRepositoriesBasePath";
+    private const string _gitExecutionTimeoutConfigurationKeyString = "GitExecutionTimeout";
 
-        if (!success)
-        {
-            // CheckPrerequisites failed, this is a server issue
-            context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
-            return;
-        }
+    private const string _gitRepoPathTemplate = @"{0}\wcf{1}";
 
-        success = CleanupMergedBranches(result);
+    public void ProcessRequest(HttpContext context)
+    {
+        StringBuilder result = new StringBuilder();
 
+        // Read configuration
+        bool success = ReadConfiguration(result);
         if (!success)
         {
-            // CleanupMergedBranches failed, this is a server issue
+            // ReadConfiguration failed, this is a server issue 
             context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
+            context.Response.Write(HttpUtility.HtmlEncode(result.ToString()));
             return;
         }
 
-        // Parameter verification
+        // Check inputs 
+        // We expect one of the following to be specified: 
+        // - A numerical repo ID
+        // - A branch name OR a PR number (not both)
+        //
+        // We will use Git to validate and verify that the branch exists if specified
 
         success = false;
 
+        string repoString = context.Request.QueryString["repo"];
         string branchString = context.Request.QueryString["branch"];
         string prString = context.Request.QueryString["pr"];
 
+        uint repoId;
+        if (string.IsNullOrWhiteSpace(repoString) || !uint.TryParse(repoString, out repoId))
+        {
+            context.Response.StatusCode = 400;
+            context.Response.Write(string.Format("The repo ID specified, '{0}', is invalid <br/>", repoString ?? "unspecified"));
+            return;
+        }
+
         if (string.IsNullOrWhiteSpace(prString) && string.IsNullOrWhiteSpace(branchString))
         {
             context.Response.StatusCode = 400;
-            context.Response.Write("No PR or branch specified");
+            context.Response.Write("No PR or branch specified. Specify either a PR or branch <br/>");
```

Comment:
Nit: could expand error message to tell them how to do this, such as name of query parameter.  Not blocking or required.